service-base/src/main/java/org/eclipse/hono/servicebase/HonoServiceBase.java

```diff
@@ -21,36 +21,47 @@
 
 import java.util.Objects;
 
-public abstract class HonoServerBase extends AbstractVerticle {
+public abstract class HonoServiceBase extends AbstractVerticle {
 
-    private static final Logger LOG = LoggerFactory.getLogger(HonoServerBase.class);
+    private static final Logger LOG = LoggerFactory.getLogger(HonoServiceBase.class);
 
     protected HonoConfigProperties config = new HonoConfigProperties();
 
     protected int port         = Constants.PORT_UNCONFIGURED;
     protected int insecurePort = Constants.PORT_UNCONFIGURED;
 
+    /**
+     * Define the default value for the TLS port number in the implementing server class (e.g. 5671 for AMQPS 1.0).
+     * @return The default value for the TLS port.
+     */
     protected abstract int getPortDefaultValue();
+    /**
+     * Define the default value for the unencrypted port number in the implementing server class (e.g. 5672 for AMQP 1.0).
+     * @return The default value for the unencrypted port.
+     */
     protected abstract int getInsecurePortDefaultValue();
 
-    public int getPort() {
+    /**
+     * Return the port number of the TLS port.
```

Comment:
What I like about that is that we do not need to update the actual port in a callback when the server is running (although probably easy), but still can ask the server itself as soon as it is running.

But since we eliminate the getters for the ServiceBase class (declaring them abstract), we need another way to test the determination phase.
Could define a method "boolean isPortDeterminedTo(int portNr)". WDYT?